NumLib/ODESolver/NonlinearSolver-impl.h

```diff
@@ -146,21 +174,53 @@ solve(Vector &x)
 
         // std::cout << "  J:\n" << Eigen::MatrixXd(J) << std::endl;
 
-        if (!_linear_solver.solve(J, res, minus_delta_x)) {
+        bool iteration_succeeded = _linear_solver.solve(J, res, minus_delta_x);
+
+        if (!iteration_succeeded)
+        {
             ERR("The linear solver failed.");
-            BLAS::axpy(x, -_alpha, minus_delta_x);
-            success = false;
+        }
+        else
+        {
+            // TODO could be solved in a better way
```

Comment:
Since there are many todos (in this PR also the merged PR), I think it is better to use doxygen command ``\todo``, which can generates a todo list, to remind one what have to do.